Lib/pathlib.py

```diff
@@ -1193,18 +1236,254 @@ def cwd(cls):
         # os.path.abspath('.') == os.getcwd().
         return cls().absolute()
 
+    def expanduser(self):
+        """ Return a new path with expanded ~ and ~user constructs
+        (as returned by os.path.expanduser)
+        """
+        self._unsupported("expanduser")
+
     @classmethod
     def home(cls):
-        """Return a new path pointing to the user's home directory (as
-        returned by os.path.expanduser('~')).
+        """Return a new path pointing to expanduser('~').
         """
         return cls("~").expanduser()
 
+    def readlink(self):
+        """
+        Return the path to which the symbolic link points.
+        """
+        self._unsupported("readlink")
+
+    def resolve(self, strict=False):
+        """
+        Make the path absolute, resolving all symlinks on the way and also
+        normalizing it.
+        """
+        if self._resolving:
+            return self
+        try:
+            path = self.absolute()
+        except UnsupportedOperation:
+            path = self
+
+        def split(path):
+            return path._from_parsed_parts(path.drive, path.root, []), path._tail[::-1]
```

Comment:
I've added a `PathBase._split_stack()` method rather than a top-level function. Not too sure about naming/placement, let me know if you think module-level makes more sense.